onnxruntime/core/providers/coreml/model/model.mm

```diff
@@ -395,9 +395,14 @@ Status Predict(const std::unordered_map<std::string, OnnxTensorData>& inputs,
       compiled_model_path_ = [compileUrl path];
 
       MLModelConfiguration* config = [[MLModelConfiguration alloc] init];
-      config.computeUnits = (coreml_flags_ & COREML_FLAG_USE_CPU_ONLY)
-                                ? MLComputeUnitsCPUOnly
-                                : MLComputeUnitsAll;
+      if (coreml_flags_ & COREML_FLAG_USE_CPU_ONLY) {
+        config.computeUnits = MLComputeUnitsCPUOnly;
+      } else if (coreml_flags_ & COREML_FLAG_USE_CPUAndGPU) {
+        config.computeUnits = MLComputeUnitsCPUAndGPU;
```

Comment:
should we also consider handling [MLComputeUnitsCPUAndNeuralEngine](https://developer.apple.com/documentation/coreml/mlcomputeunits/cpuandneuralengine?language=objc)?

Reply:
The problem this is addressing is that CoreML choosing to use ANE can result in worse perf for a model with large data sizes. Really CoreML compute unit selection should do a better job, but we don't control that.

I'm not aware of a scenario where excluding GPU would be of benefit, so I suggested we don't handle that as there's no known use case.